src/app/projects/components/project-settings/components/embeddings/embeddings.component.html

```diff
@@ -166,12 +166,17 @@
         corresponding
         tensors!</p>
 </kern-modal>
-
-
-<kern-modal [isOpen]="settingModals.embedding.create.open" [modalBoxStyle]="{'overflow-x':'visible'}" closeButton="X"
+<kern-modal [isOpen]="settingModals.embedding.create.open"
+    [modalBoxStyle]="{'overflow-x':'visible', 'max-width':'80%', 'width': 'fit-content'}" closeButton="X"
     [acceptButton]="{
     buttonCaption: 'Add embedding',
-    disabled: settingModals.embedding.create.blocked || settingModals.embedding.create.embeddingCreationFormGroup?.get('embeddingHandle').value == '' || !(useableTextAttributes && settingModals.embedding.create.embeddingCreationFormGroup),
+    disabled: settingModals.embedding.create.blocked || 
+    !(useableTextAttributes && settingModals.embedding.create.embeddingCreationFormGroup) || 
+    !settingModals.embedding.create.embeddingCreationFormGroup.get('termsAccepted').value ||
+    (settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value == PlatformType.HUGGING_FACE && settingModals.embedding.create.embeddingCreationFormGroup.get('model').value == null) ||
+    (settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value == PlatformType.OPEN_AI && (settingModals.embedding.create.embeddingCreationFormGroup.get('model').value == null || settingModals.embedding.create.embeddingCreationFormGroup.get('apiToken').value === null)) ||
+    (settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value == PlatformType.COHERE && settingModals.embedding.create.embeddingCreationFormGroup.get('apiToken').value === null) ||
+    (settingModals.embedding.create.embeddingCreationFormGroup.get('platform').value == PlatformType.PYTHON && settingModals.embedding.create.embeddingCreationFormGroup.get('model').value == null),
```

Comment:
this feels like way to much in the HTML file conditioning so it's not really readable. Can we create a disabled variable and use that & change it whenever necessary? 

I'm thinking of a function "isDisabled" that sets this condition to the disabled variable and is called on change of either of the values so it's more readable. Not linking a function in the HTML :)